ChromeExtension/js/apiKeyViewModel.js

```diff
@@ -1,75 +1,64 @@
-function apiKeyViewModel() {
-
-
-    var self = this;
-    self.apiKey = ko.observable('');
-    self.apiSecret = ko.observable('');
-    self.showBackLink = ko.observable(false);
-    self.showHelpLink = ko.observable(true);
-    self.newInstall = ko.observable(true);
-    self.UrlApiKeys = ko.observable('');
-
-
-    self.apiKey.subscribe(function (newValue) {
-        localStorage.setItem("apiKey", newValue);
-    });
-
-    self.apiSecret.subscribe(function (newValue) {
-        localStorage.setItem("apiSecret", newValue);
-    });
-
-
-    self.loadKey = function () {
-        var ak = localStorage["apiKey"];
-        if (typeof ak !== 'undefined') {
-            self.apiKey(ak);
-        }
-        var asa = localStorage["apiSecret"];
-        if (typeof asa !== 'undefined') {
-            self.apiSecret(asa);
-        }
+var apiKeyViewModel = {
+    apiKey: ko.observable(''),
+    apiSecret: ko.observable(''),
+    showBackLink: ko.observable(false),
+    showHelpLink: ko.observable(true),
+    newInstall: ko.observable(true),
+    UrlApiKeys: ko.observable('')
+};
+
+apiKeyViewModel.apiKey.subscribe(function (newValue) {
+    chrome.storage.local.set({"apiKey": newValue});
+});
 
-    }
+apiKeyViewModel.apiSecret.subscribe(function (newValue) {
+    chrome.storage.local.set({"apiSecret": newValue});
+});
 
-    if (localStorageHasValue("defaultGroup")) {
-        self.showBackLink(true);
-        self.showHelpLink(false);
-        self.newInstall(false);
-        self.UrlApiKeys = ('https://my.geni.us/tools#api-section');
+chrome.storage.local.get(["defaultGroup"]).then((group) => {
+    if (group != null && group != "") {
+        apiKeyViewModel.showBackLink(true);
+        apiKeyViewModel.showHelpLink(false);
+        apiKeyViewModel.newInstall(false);
+        apiKeyViewModel.UrlApiKeys = ('https://my.geni.us/tools#api-section');
     }
+});
 
-    self.saveKeys = function () {
-        var Empty = "";
-        if (self.apiKey() !== null && self.apiSecret() !== null && self.apiKey() !== Empty && self.apiSecret() !== Empty) {
-            localStorage.setItem("apiKey", self.apiKey());
-            localStorage.setItem("apiSecret", self.apiSecret());
-
-
-        } else {
-            $("#dialog").dialog({
-                draggable: false,
-                modal: true
-            });
-
-            $("#networkError").html("Oops! Those keys don't appear to be right. Please double check your API Key and Secret.");
-
+apiKeyViewModel.loadKey = function() {
+    chrome.storage.local.get(["apiKey"]).then((result) => {
```

Comment:
yes, opening the extension in Incognito mode does save the API keys and user preferences (default group, default domain) so when opening the extension in private browsing, user will be immediately greeted with this UI instead of having to input their API keys again:
![image](https://github.com/georiot/client-apps/assets/25525058/90c1e593-4981-4811-8e6e-3beecfecd951)